dags/fetcher.py

```diff
@@ -1,61 +1,132 @@
-from datetime import datetime, timedelta
-import time
+import json
+import typing
+import logging
 
-# The DAG object; we'll need this to instantiate a DAG
+from datetime import datetime, timedelta
+ 
 from airflow import DAG
-
-# Operators; we need this to operate!
 from airflow.operators.python import PythonOperator
 from airflow.providers.postgres.operators.postgres import PostgresOperator
 
-# These args will get passed on to each operator
-# You can override them on a per-task basis during operator initialization
+
 default_args = {
     'owner': 'airflow',
     'depends_on_past': False,
     'email': ['airflow@example.com'],
-    'email_on_failure': False,
+    'email_on_f0ilure': False,
     'email_on_retry': False,
     'retries': 1,
-    'retry_delay': timedelta(minutes=5),
+    'retry_delay': timedelta(minutes=1),
 }
+
 with DAG(
-        'fetcher',
-        default_args=default_args,
-        description='To fetch the weather data',
-        schedule_interval=timedelta(minutes=5),
-        start_date=datetime(2021, 1, 1),
-        catchup=False,
-        tags=['take-home'],
+  'fetcher',
+  default_args=default_args,
+  description='To fetch the weather data',
+  start_date=datetime(2021, 1, 1, 0, 0, 0),
+  schedule_interval="0 */2 * * *",
+  catchup=False,
+  tags=['take-home'],
 ) as dag:
 
-    # @TODO: Add your function here. Example here: https://airflow.apache.org/docs/apache-airflow/stable/_modules/airflow/example_dags/example_python_operator.html
-    # Hint: How to fetch the weather data from OpenWeatherMap?
-    def my_sleeping_function(random_base):
-        """This is a function that will run within the DAG execution"""
-        time.sleep(random_base)
+    create_raw_dataset = PostgresOperator(
+        task_id="create_raw_dataset",
+        sql="sql/create_raw_current_weather_tbl.sql",
+    )
+
+    def run_extract(api_key: str, city_ids: typing.List[str], save_to_file: bool = False) -> typing.List[typing.Tuple]:
+        """
+        Pull payload from OpenWeatherAPI from a list of city IDs.
+        """
+        from openweather.op.extract import OpenWeatherMapExtractor
+
+        extractor = OpenWeatherMapExtractor(api_key=api_key)
+        rows = []
+        for city_id in city_ids:
+            payload = extractor.get_current_weather(city_id=city_id)
+            if payload:
+                # At this stage (didn't implement) we could implement
+                # some kind of schema validation (pydantic, etc.)
```

Comment:
This and some other comments are giving away how one could get more browny points, is this intentional?